analysisservice/src/main/java/nu/yona/server/analysis/service/ActivityCacheService.java

```diff
@@ -16,40 +15,34 @@
 import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import nu.yona.server.analysis.entities.Activity;
 import nu.yona.server.analysis.entities.DayActivity;
-import nu.yona.server.analysis.entities.WeekActivity;
 
 @Service
 // Day activities are only used in the analysis engine service, so a local cache suffices for as long as we do not
 // scale out the analysis engine service.
 @CacheConfig(cacheManager = "localCache")
 public class ActivityCacheService
 {
-	@Cacheable(value = "dayActivities", key = "{#userAnonymizedID,#goalID}")
+	@Cacheable(value = "activities", key = "{#userAnonymizedID,#goalID}")
```

Comment:
This is not a collection of activities, but only the last one, right? So please rename this into ''lastActivity''.